.github/workflows/ci-tests.yml

```diff
@@ -189,4 +189,22 @@ jobs:
       uses: codecov/codecov-action@v1
       with:
         files: ./coverage.xml
+  pip-test-pull-request:
+    # Test that using pip install works as we've missed
+    # some dependencies in the past - see #1842
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    name: pip install tests
+    steps:
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.10'
+    - name: Install dependencies
+      run: |
+        pip install git+https://github.com/${{ github.event.pull_request.head.repo.full_name }}@${{ github.head_ref }}
+    - name: Run the tests
+      run: |
+        sqlfluff --version
```

Comment:
Should we run a command that exercises more of SQLFluff's functionality? Python is a dynamic language, so it's not going to check all the `import` statements, only those in modules that get loaded.

Reply:
Added a couple of more simple tests:

```yml
    - name: Run the version test
      run: |
        sqlfluff --version
    - name: Run a simple select parse test
      run: |
        echo "select 1" | sqlfluff parse -
    - name: Run a simple select lint test
      run: |
        echo "select 1" | sqlfluff lint -
```